Add AMD GPU ROCm/HIP backend

Looong01 opened the pull request:
All test passed. It could be merged.

Looong01 committed:
Add hipCollections submodule

Looong01 committed:
Add rocm backend

Looong01 committed:
Fix bugs

Looong01 committed:
Fix bugs

Looong01 committed:
Update Readme

Looong01 committed:
Update comments

Looong01 committed:
Check again

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Looong01 committed:
Update changlog for rocm

Looong01 committed:
Add external repo for rocm backend

Looong01 committed:
[ROCm] import third_party/libhipcxx automatically

Looong01 committed:
Fix bugs

gpicciuca commented:
Bumping! Why is there no feedback from the maintainers? No interest in supporting ROCm? Would love to see this merged and have proper wheels released.

akihironitta committed:
Merge branch 'master' into master

gpicciuca commented:
> > Why is there no feedback from the maintainers? No interest in supporting ROCm?
> 
> @gpicciuca I think we would definitely be happy to add support for ROCm, but we don't have access to AMD GPUs and don't have good ways for us to test ROCm builds at the moment. The number of AMD GPU users is still much smaller than that of NVIDIA GPUs users, and given the fact that we're a very small team, I don't see it feasiable for us to keep maintaining ROCm builds in coming months or year(s) (unless, e.g., AMD team can provide assistance).

I could probably lend a hand here as I own an RX 9070 XT. Using Windows 10 + WSL2 with ROCm 6.4 installed according to AMD's instructions.

Also, have you tried pinging some AMD dev? I think they are active in some Pytorch repository.. so they could potentially contribute to PyG as well i think.. but yea, don't take my words for granted as I don't work for AMD

Looong01 commented:
Sorry, the matrix multiple kernel still need to be implement by composable kernel(ck) library instead of hipblas_lt. It is hard to learn ck.

Copilot committed:
Initial plan

Copilot committed:
Implement matmul_kernel using Composable Kernel (CK) for ROCm

- Create matmul_kernel_ck.hip using CK's DeviceGroupedGemm_Xdl
- Remove props.major/props.minor version checks (not needed for AMD GPUs)
- Remove use_tf32 logic (not applicable to AMD GPUs)
- Implement only DefaultGemmKernel_FP32
- Update CMakeLists.txt to add composable_kernel dependency
- Remove old hipblaslt-based matmul_kernel.hip (replaced by CK version)

Co-authored-by: Looong01 <69568351+Looong01@users.noreply.github.com>

Copilot committed:
Fix StreamConfig namespace - use ck::StreamConfig

Co-authored-by: Looong01 <69568351+Looong01@users.noreply.github.com>

Copilot committed:
Add comment about kernel parameters following CUDA CUTLASS pattern

Co-authored-by: Looong01 <69568351+Looong01@users.noreply.github.com>

Copilot committed:
Update matmul_kernel_ck.hip to use official AMD CK API with GemmDesc

- Use ck::tensor_operation::device::GemmDesc structure per official API
- Include device_grouped_gemm.hpp header for GemmDesc definition
- Add NumDTensor constexpr for clarity
- Update MakeArgument call to use vector<GemmDesc> instead of separate vectors
- Add reference comments to official CK documentation
- Improve documentation comments explaining kernel parameters

Co-authored-by: Looong01 <69568351+Looong01@users.noreply.github.com>

Looong01 committed:
Merge pull request #1 from Looong01/copilot/implement-matmul-composable-kernel

Replace hipBLASLt matmul with Composable Kernel implementation for ROCm

Commit:
Upgrade CK implement

Commit:
update readme

Commit:
update gitmodules

Commit:
Remove hipCollections and libhipcxx submodules

Commit:
Add hipCollections and libhipcxx as submodules

Commit:
Upgrade CK backend and fix bugs

Commit:
Fix bugs

Commit:
Update README

Commit:
Resolve conflicts

Commit:
Resolve conflicts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Looong01 committed:
Merge branch 'master' into master

Looong01 committed:
Resolve conflicts

Looong01 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Looong01 committed:
Add exclusions for libcudart and libnvrtc in pyproject

Looong01 committed:
Merge branch 'master' into master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Looong01 committed:
Update to v0.8.0

Looong01 commented:
Have Updated to v0.8.0
